les/serverpool.go

```diff
@@ -76,16 +76,24 @@ var (
 	sfSelected      = utils.NewNodeStateFlag("selected", false, false)
 	sfDialed        = utils.NewNodeStateFlag("dialed", false, false)
 	sfConnected     = utils.NewNodeStateFlag("connected", false, false)
-	sfRedialWait    = utils.NewNodeStateFlag("redialWait", false, true)
+	sfRedialWait    = utils.NewNodeStateFlag("redialWait", false, false)
 	sfAlwaysConnect = utils.NewNodeStateFlag("alwaysConnect", false, false)
 
-	keepNodeRecord   = []*utils.NodeStateFlag{sfDiscovered, sfHasValue, sfAlwaysConnect}
+	// keepNodeRecord is a group of state flags. If any
+	// flag is set, then the node record should be kept.
+	keepNodeRecord = []*utils.NodeStateFlag{sfDiscovered, sfHasValue, sfAlwaysConnect}
+
+	// disableSelection is a group of state flags. If any
+	// flag is set, then the node is considered as selected
+	// and won't be selected for further network activities.
 	disableSelection = []*utils.NodeStateFlag{sfSelected, sfDialed, sfConnected, sfRedialWait}
 
 	errInvalidField = errors.New("invalid field type")
 
 	sfiEnr = utils.NewNodeStateField("enr", reflect.TypeOf(&enr.Record{}), keepNodeRecord,
 		func(field interface{}) ([]byte, error) {
+			// todo(rjl493456442) enr record can only
+			// be RLP-encoded when the record is signed
```

Comment:
Can we always ensure the node record from the network is signed? need to check in the p2p level.